test/integration/setup-gitlab/bootstrap_gitlab_environment_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bootstrap_gitlab
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/shell"
+	"github.com/terraform-google-modules/enterprise-application/test/integration/testutils"
+	gitlab "gitlab.com/gitlab-org/api/client-go"
+)
+
+func getTokenFromSecretManager(t *testing.T, secretName string, secretProject string) (string, error) {
+	t.Log("Retrieving secret from secret manager.")
+	cmd := fmt.Sprintf("secrets versions access latest --project=%s --secret=%s", secretProject, secretName)
+	args := strings.Fields(cmd)
+	gcloudCmd := shell.Command{
+		Command: "gcloud",
+		Args:    args,
+		Logger:  logger.Discard,
+	}
+	return shell.RunCommandAndGetStdOutE(t, gcloudCmd)
+}
+
+// connects to a Google Cloud VM instance using SSH and retrieves the logs from the VM's Startup Script service
+func readLogsFromVm(t *testing.T, instanceName string, instanceZone string, instanceProject string) (string, error) {
+	args := []string{"compute", "ssh", instanceName, fmt.Sprintf("--zone=%s", instanceZone), fmt.Sprintf("--project=%s", instanceProject), "--command=journalctl -u google-startup-scripts.service -n 20"}
+	gcloudCmd := shell.Command{
+		Command: "gcloud",
+		Args:    args,
+	}
+	return shell.RunCommandAndGetStdOutE(t, gcloudCmd)
+}
+
+func TestBootstrapGitlabVM(t *testing.T) {
+	// Retrieve output values from test setup
+	setup := tft.NewTFBlueprintTest(t,
+		tft.WithTFDir("../../setup"),
+	)
+	url := setup.GetStringOutput("gitlab_url")
+	gitlabSecretProject := setup.GetStringOutput("gitlab_secret_project")
+	gitlabSecretProjectNumber := setup.GetStringOutput("gitlab_project_number")
+	gitlabPersonalTokenSecretName := setup.GetStringOutput("gitlab_pat_secret_name")
+	gitlabWebhookSecretId := setup.GetStringOutput("gitlab_webhook_secret_id")
+	instanceName := setup.GetStringOutput("gitlab_instance_name")
+	instanceZone := setup.GetStringOutput("gitlab_instance_zone")
+	gitlabTokenSecretId := fmt.Sprintf("projects/%s/secrets/%s", gitlabSecretProjectNumber, gitlabPersonalTokenSecretName)
+
+	// Periodically read logs from startup script running on the VM instance
+	for count := 0; count < 10; count++ {
```

Comment:
Just FYI, as you are already using `github.com/gruntwork-io/terratest/modules`, I'd suggest looking at: [modules/retry](https://github.com/gruntwork-io/terratest/blob/main/modules/retry/retry.go)